moto/cloudformation/models.py

```diff
@@ -254,9 +258,7 @@ def _create_resource_map(self):
         return resource_map
 
     def _create_output_map(self):
-        output_map = OutputMap(self.resource_map, self.template_dict, self.stack_id)
-        output_map.create()
```

Comment:
The full workflow looked like this:
  - Create FakeStack:
      - Initialize resource map
      - Create output map
 - Persist a reference to the FakeStack
 - create resources as defined in resource map

However:
When the OutputMap contained a reference to a resource (using `fn:getatt`), creating the OutputMap would also create the resource - as the resource needs to be created before we can retrieve the attribute.
This could lead to the scenario where a resource is created, but creation of the OutputMap ultimately fails because we're trying to `fn:getatt` an unknown attribute.. As the OutputMap was created in the FakeStack-constructor, the creation of the FakeStack also fails, and the resource itself is dangling.